viadot/sources/hubspot.py

```diff
@@ -0,0 +1,171 @@
+    Connector allows to pull data in three ways
+        - using base API for crm schemas as an endpoint (eg. "contacts", ""line_items", "deals", ...)
+        - using full url as endpoint
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        DEFAULT_CREDENTIALS = local_config.get("HUBSPOT")
+        self.credentials = kwargs.pop("credentials", DEFAULT_CREDENTIALS)
+        if self.credentials is None:
+            raise CredentialError("Missing credentials.")
+        super().__init__(*args, credentials=self.credentials, **kwargs)
+
+        self.API_ENDPOINT = self.credentials.get("URL")
+
+    def clean_special_characters(self, value: str = None) -> str:
+        """
+        Function that maps special characters from ASCII to HTML5 equivalents.
+
+        Args:
+            value (str, optional): Value for decoding. Defaults to None.
+
+        Returns:
+            str: Value after decoding to HTML5.
+        """
+
+        mapping = {
+            "!": "%21",
+            '"': "%22",
+            "#": "%23",
+            "$": "%24",
+            "%": "%25",
+            "&": "%26",
+            "(": "%28",
+            ")": "%29",
+            "*": "%2A",
+            "+": "%2B",
+            ",": "%2C",
+        }
+
+        for key in mapping.keys():
+            if key in value:
+                value = value.replace(key, mapping[key])
+
+        return value
+
+    def get_properties_url(self, endpoint: str = None) -> str:
+        """
+        Function that generates URL with all properties ( = detailed information about all avaliable columns for defined endpoint (schema)).
+
+        Args:
+            endpoint (str, optional): Hubspot schema from crm objects (eg. contacts, line_items, ...). Defaults to None.
+
+        Returns:
+            str: Url with properties for defined endpoint.
+        """
+
+        url = f"{self.API_ENDPOINT}/properties/v2/{endpoint}/properties"
+        return url
+
+    def get_api_url(
+        self,
+        endpoint: str = None,
+        filters: Dict[str, Any] = None,
+        properties: List[Any] = None,
+    ) -> str:
+
+        """
+        Function that generates full url for Hubspot API with defined parametrs.
+        Args:
+            endpoint (str, optional): Schema or full url. Defaults to None.
+            filters (Dict[str, Any], optional): Filters defined for the API body in specific order. Defaults to None.
+            properties (List[Any], optional): List of user-defined columns to be pulled from the API. Defaults to None.
+
+        Returns:
+            str: Generated url passed to Hubspot API
+        """
+
+        if self.API_ENDPOINT in endpoint:
+            url = endpoint
+        else:
+            if filters:
+                url = (
+                    f"{self.API_ENDPOINT}/crm/v3/objects/{endpoint}/search/?limit=100&"
```

Comment:
You are checking `if filters` but you are not using filters here. Is it correct?

Reply:
Logic has been changed:
`                if filters:
                    url = (
                        f"{self.base_url}/crm/v3/objects/{endpoint}/search/?limit=100&"
                    )
                else:
                    url = f"{self.base_url}/crm/v3/objects/{endpoint}/?limit=100&"`
if filters are defined -> add  `search` and `method = "POST"`, it's defined in the task class